python/tests/conftest.py

```diff
@@ -1277,8 +1277,8 @@ def adlfs_fsspec_fileio(request: pytest.FixtureRequest) -> Generator[FsspecFileI
     azurite_account_key = request.config.getoption("--adlfs.account-key")
     azurite_connection_string = f"DefaultEndpointsProtocol=http;AccountName={azurite_account_name};AccountKey={azurite_account_key};BlobEndpoint={azurite_url}/{azurite_account_name};"
     properties = {
-        "connection_string": azurite_connection_string,
-        "account_name": azurite_account_name,
+        "adlfs.connection-string": azurite_connection_string,
+        "adlfs.account-name": azurite_account_name,
```

Comment:
What would be your suggestion? I noticed that you can't set `endpoint` using the constructor. This can only be set using the connection string.

Reply:
@Fokko 

I believe the simplest approach would be to utilize connection-string, as it can accommodate various possibilities such as a SAS Token. 

For example Trino/Starburst uses account-name and access-key  Azure Storage.
Check please Hive Connector with Azure Storage in Trino as reference.
https://trino.io/docs/current/connector/hive-azure.html#

Based on the information provided, my recommendation would be to allow for both a connection-string or an account-name and account-key to be used. This would provide flexibility for users who may prefer one method over the other.


For the Azure Storage URL's there are following possibilites. I guess this is for the other PR relevant.

- abfs
- abfss
- wasb 
- wasbs
- adl

Reply:
When I wrote that code, I only supported what I could test, which is Azure Gen2 Datalake (or Storage Account).  Gen2 is nice since it provides file system semantics, file-level security, and scale combined with blob storage benefits.  That's why `pyiceberg.io.fsspec` only uses a `AzureBlobFileSystem` instance (which supports `abfs` and `az` protocols).  Could that be extended to support Gen1?  Probably since fsspec supports it, but I wasn't in a position to test it.

As for the required configuration, Azure supports multiple ways to authenticate and that's why I left it open, to support all possibilites:
```python
def _adlfs(properties: Properties) -> AbstractFileSystem:
    from adlfs import AzureBlobFileSystem

    fs = AzureBlobFileSystem(**properties)
    return fs
```
Authentication is handled by `AzureBlobFileSystem`, who delegates to [Azure Identity](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/identity/azure-identity).

But I understand that this does not provide good documentation for a user, so I'm all for listing configuration options.  The question is do we want to limit those options to only a few (so we code checks as @alaturqua suggested), or we provide examples in the documentation and we delegate validation to `AzureBlobFileSystem`?

It can get tricky if we want to code checks for every possibility.  As an example, here is the documentation for `account_key` and `sas_token`:
```
    account_key: str
        The storage account key. This is used for shared key authentication.
        If any of account key, sas token or client_id is specified, anonymous access
        will be used.
    sas_token: str
        A shared access signature token to use to authenticate requests
        instead of the account key. If account key and sas token are both
        specified, account key will be used to sign. If any of account key, sas token
        or client_id are specified, anonymous access will be used.
```
I'm not saying it cannot be done, but it requires attention.

I don't know what is Iceberg philosophy when it comes to using an external library, i.e. should Iceberg retain full control or delegation is fine?

Reply:
Thanks for the responses @alaturqua and @cccs-eric 

> The implementation could check and force to use one of them either connection-string or account-name + accunt-key.

This is also how it originally was intended, I'm all for it.

> For the Azure Storage URL's there are following possibilites. I guess this is for the other PR relevant.

I've added those, did you have a chance of testing these?

> I don't know what is Iceberg philosophy when it comes to using an external library, i.e. should Iceberg retain full control or delegation is fine?

We are liberal in the options that we accept, but we want to unify them. For example, for S3 there are two FileIO's, PyArrow and S3FS. You can switch between them without having to change your configuration. Therefore we have to do some mapping between the names.

Reply:
I think we're on the same page. I'll merge this PR now, and feel free to open up any follow-up PRs if you feel like it should be done differently.